src/thread_event.c

```diff
@@ -18,6 +18,16 @@ static void thread_##event##_event_handler(tsd_t *tsd);
 ITERATE_OVER_ALL_EVENTS
 #undef E
 
+static void
+thread_tcache_gc_event_handler(tsd_t *tsd) {
+	assert(opt_tcache);
+	assert(tcache_gc_event_wait_get(tsd) == 0U);
+	if (TCACHE_GC_INCR_BYTES > 0U && tcache_available(tsd)) {
+		thread_tcache_gc_event_update(tsd, TCACHE_GC_INCR_BYTES);
```

Comment:
Leave the update outside of the branch? tcache can be switched on/off per thread; if we don't reset it, we might never handle the event again?